pybandits/quantitative_model/base.py

```
@@ -0,0 +1,249 @@
+
+    @classmethod
+    def _deserialize_callable(cls, value):
+        """Deserialize a callable from its string representation if needed."""
+        if isinstance(value, str):
+            if value.startswith("functools.partial"):
+                inner_func_split = "(".join(value.split("(")[1:]).split(",")
+                # Extract function and arguments from pattern: functools.partial(func_name, args, kwargs)
+                func_str = ",".join(inner_func_split[:-2]).strip()
+                func = cls._deserialize_function(func_str)
+                args_str = ")".join(",".join(inner_func_split[-2:]).split(")")[:-1]).strip()
+                args_parts = eval(args_str) if args_str else ((), {})
```

Ruff (0.15.15)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)